### ISSUE TITLE
Do you accept donations??

### ISSUE THREAD

**Issue by ameykshirsagar** [civetweb/civetweb#523]
Hi,
Is there a way to give some sort of donation to the project??
Because we at Arcoiris Labs are quite impressed with the work you did and we are willing to integrate civetweb in our product.
So just to say thanks, can I get the details on donation options??

Thanks
Amey K.

**Comment by bel2125** [comment]
Maintaining CivetWeb is indeed a certain effort, and some donation is welcome :smile:. It‘s not required by the license, but it‘s certainly appreciated.

I did not do it for the money, so I did not really evaluate different donation systems yet.
Every now and then, someone asks me by mail (my username at gmail.com) for a private consulting on their project, and if it was something that needs some investigation, we agreed on some compensation by PayPal. That’s not exactly a donation, but PayPal is the only system I have been using for online projects yet. Do you have any other suggestions?

**Comment by ameykshirsagar** [comment]
Hello @bel2125 ,
We are developing a mobile application framework from ground up and we want to integrate civetweb in our framework. Although the licence doesn't insists on paying the maintainer(s), we know the efforts in maintaining a software as I at my company @arcoirislabs do. Being a self funded startup we would love to donate as much as we can to this project & later be a part of it to improve this project.
Paypal works great for donations, so I personally don't think any other options are viable to implement the donation part. You can look [here](https://github.com/patharanordev/donate-in-git) 
I shall mail you the details of the donation as soon as we hit the first wave of sales.
Thanks for the response and cheers!!
Sorry to raise this discussion in issues section

**Comment by bel2125** [comment]
Thanks for the link, I figured out how to create a proper button.

Badge style:

[![Donate](https://img.shields.io/badge/Donate-PayPal-green.svg)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=88ZLXZ6U77GJU)

PayPal style:

[![paypal](https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=88ZLXZ6U77GJU)

Now I need to find some place to put it, where it can be found without being too obtrusive.

**Comment by ameykshirsagar** [comment]
You can also try Gratipay

**Comment by bel2125** [comment]
Now Gratipay is available as well:
[![Gratipay](https://img.shields.io/gratipay/project/CivetWeb.svg)](https://gratipay.com/CivetWeb/)
Thanks for the hint.

**Comment by bel2125** [comment]
Note: [Gratipay.com](https://gratipay.com/) does no longer exist since the end of 2017, see announcement from November 1 2017: https://gratipay.news/the-end-cbfba8f50981

PayPal still works

**Comment by DavidPDev** [comment]
Hi, I know this is very old "issue", but I would be able to pay some small recurrent donation through Patreon.
@bel2125 do you think Patreon could be an option for you?
Thank you for all your great work and efforts! 🤩

**Comment by bel2125** [comment]
Actually I had a donation link open for several years and received 0.0 during that time - not a single donation. At some conference I learned that donations, in particular recurring ones, may lead to a project being legally considered as commercial, resulting in some some legal obligations for the "supplier" (me ... some who send me patches by email ...). On the other hand, if there is only one paying "client", this may be considered as an employment in disguise.
After hearing that I removed the donation link, since I got the impression that for every 10€ donation I get I will need more than 100€ for legal consultation. I never received any payment anyway, not even enough to buy a bouquet of flowers for my wife that she deserved because once again I spent too much time at the PC in the evening. So I decided to remove the donation links some months ago, since I never did this for the money anyway.
